devicetypes/smartthings/zigbee-smoke-sensor.src/zigbee-smoke-sensor.groovy

```diff
@@ -85,7 +85,9 @@ def parse(String description) {
 def parseAttrMessage(String description){
 	def descMap = zigbee.parseDescriptionAsMap(description)
 	def map = [:]
-	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
+	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.attrId == "0020" && descMap.commandInt != 0x07 && descMap.value) {
+		map = getBatteryResult(Integer.parseInt(descMap.value, 16))
+	} else if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
 		map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
```

Comment:
Given my comment about using named constants, we should be able to change this logic up a little:
```
	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
		if (descMap.attrInt == BATTERY_VOLTAGE_ATTR) {
			map = getBatteryResult(Integer.parseInt(descMap.value, 16))
		} else {
			map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
		}
	}
```

Reply:
Makes sense Donald. Added the changes.